### PR TITLE
Adding a private indicator for the activation of the terminal logger

### PR DESCRIPTION
**Pull request by MichalPavlik**
Adding a private indicator for the activation of the terminal logger - it should be used only by testing platform team temporarily.

### CODE REVIEW

File: src/MSBuild/XMake.cs [hunk 1]
```diff
@@ -2516,6 +2516,11 @@ private static bool GatherAutoResponseFileSwitchesFromFullPath(string autoRespon
 #endif
 
             bool useTerminalLogger = ProcessTerminalLoggerConfiguration(commandLineSwitches, out string aggregatedTerminalLoggerParameters);
+
+            // This is temporary until we can remove the need for the environment variable.
+            // DO NOT use this environment variable for any new features as it will be removed without further notice.
+            Environment.SetEnvironmentVariable("_MSBUILDTLENABLED", useTerminalLogger ? "1" : "0");
```
Comment (JanKrivanek) [inline src/MSBuild/XMake.cs:2522]: Just a curiosity - why is this underscore prefixed? We have other 'not supposed to be used' env vars - all starts with 'MSBUILD'

**Review by JanKrivanek — APPROVED** [review]
Looks good

File: src/MSBuild/XMake.cs (hunk 1 above)
Comment (MichalPavlik) [reply src/MSBuild/XMake.cs:2522]: It felt right to do it as we are doing it for properties and additionally, this variable should be removed in the future :) Let me know if you insist to remove it.

Comment (rainersigwald) [reply src/MSBuild/XMake.cs:2522]: Feels right to me too.

Comment (JanKrivanek) [reply src/MSBuild/XMake.cs:2522]: Oh no - I was just curious.
Let's get this in

**Pull request merged**